### PR TITLE
Update gfx and naga to gfx-22

### PR DESCRIPTION
**Pull request by Sineaggi**
In order to support cases where enums in webgpu-headers and wgpu might diverge, I've added a new match for more direct mapping between two variants.

### CODE REVIEW

File: examples/capture/main.c [hunk 1]
```diff
@@ -33,7 +33,7 @@ int main(
                 },
                 .maxBindGroups = 1,
                 .label = "Device",
-                .tracePath = NULL,
+                .tracePath = "build/",
```
Comment (kvark) [inline examples/capture/main.c:36]: leftover?

File: examples/triangle/main.c [hunk 2]
```diff
@@ -123,7 +123,7 @@ int main()
                 },
                 .maxBindGroups = 1,
                 .label = "Device",
-                .tracePath = NULL,
+                .tracePath = "build/",
```
Comment (kvark) [inline examples/triangle/main.c:126]: here also. is this intentional?

File: examples/capture/main.c (hunk 1 above)
Comment (Sineaggi) [reply examples/capture/main.c:36]: was making it match the other one, maybe that was leftover? https://github.com/gfx-rs/wgpu-native/blob/master/examples/compute/main.c#L35

File: examples/triangle/main.c (hunk 2 above)
Comment (Sineaggi) [reply examples/triangle/main.c:126]: I don't mind making all of them NULL

Comment (kvark) [reply examples/triangle/main.c:126]: let's make them NULL, yes

Comment (Sineaggi) [reply examples/triangle/main.c]: Sounds good.

**Pull request merged**